cmd/podman/cp.go

```diff
@@ -210,10 +210,26 @@ func copyBetweenHostAndContainer(runtime *libpod.Runtime, src string, dest strin
 			srcPath = cleanedPath
 		}
 	}
-	glob, err = filepath.Glob(srcPath)
-	if err != nil {
-		return errors.Wrapf(err, "invalid glob %q", srcPath)
+
+	// Don't glob srcPath if it's the value of a symlink
+	// Fix for https://bugzilla.redhat.com/show_bug.cgi?id=1741709
+	fi, err := os.Lstat(srcPath)
+	switch errors.Cause(err).(type) {
+	case *os.PathError:
+		glob = append(glob, srcPath)
```

Comment:
Maybe my headache today is getting the better of me, but why do you append the glob if lstat returns a PathError (which I think is it's main error?)?  I'd thought you'd only continue on for nil, and hit default for the others?

Reply:
@TomSweeneyRedHat I wanted the code after the switch to throw the expected errors etc. Trying to keep the patch as small as possible.

Reply:
OK, I'm a bit more confused.  What I thought should happen is line 218 and 219 should disappear.  I thought if we couldn't resolve the file or symlink, it should die. That would make the patch even smaller....  Back to dig up some more tylenol.

Reply:
I would remove the glob functionality. The functionality does not seem to be mentioned in the Podman documentation and it seems the functionality is not present in Docker (please correct me if I'm wrong). 

If the functionality needs to be kept, to avoid surprises podman should only do globbing on exactly the path that the user provided on the command line. As soon as symlinks have been followed the path could look quite different. I think at this point in the code the variable `srcPath` could have some other value than what the user provided on the command line.

Reply:
@eriksjolund I made the mistake of assuming '*' was expected as it was in the code. :)

For the container vibrant_brattain:
```
$ sudo docker cp vibrant_brattain:/etc/* /tmp
Error: No such container:path: vibrant_brattain:/etc/*
```
So this is a feature we've introduced.  I'll back it out and test.

@mheon Sound good?